From 97b9c4fb032f321c4c622f6650f405a656e3905b Mon Sep 17 00:00:00 2001
From: Mykyta Biliavskyi <m.biliavskyi@samsung.com>
Date: Thu, 16 Jun 2016 14:33:32 +0300
Subject: [PATCH 55/64] Edje_edit: fix text_source_set and
 text_text_source_set.

Do not need to replace text value inside edje_edit_state_text_text_source_set,
because edje_text module will use text from source.
---
 src/lib/edje/edje_edit.c | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/src/lib/edje/edje_edit.c b/src/lib/edje/edje_edit.c
index 026ea57..9f601ce 100644
--- a/src/lib/edje/edje_edit.c
+++ b/src/lib/edje/edje_edit.c
@@ -8398,9 +8398,7 @@ edje_edit_state_text_text_source_get(Evas_Object *obj, const char *part, const c
 EAPI Eina_Bool
 edje_edit_state_text_text_source_set(Evas_Object *obj, const char *part, const char *state, double value, const char *source)
 {
-   Edje_Part_Description_Common *spd;
-   Edje_Part_Description_Text *txt, *source_txt;
-   const char *text_source;
+   Edje_Part_Description_Text *txt;
    int id_text_source;
 
    GET_PD_OR_RETURN(EINA_FALSE);
@@ -8411,16 +8409,12 @@ edje_edit_state_text_text_source_set(Evas_Object *obj, const char *part, const c
 
    if (source)
      {
-        spd = _edje_part_description_find_byname(eed, source, state, value);
-        if (!spd) return EINA_FALSE;
         txt = (Edje_Part_Description_Text *)pd;
-        source_txt = (Edje_Part_Description_Text *)spd;
 
         id_text_source = _edje_part_id_find(ed, source);
         txt->text.id_text_source = id_text_source;
 
-        text_source = source_txt->text.text.str;
-        _edje_if_string_replace(ed, &txt->text.text.str, text_source);
+        txt->text.text.str = eina_stringshare_add(NULL);
         txt->text.text.id = 0;
      }
    else
@@ -8457,7 +8451,6 @@ edje_edit_state_text_source_get(Evas_Object *obj, const char *part, const char *
 EAPI Eina_Bool
 edje_edit_state_text_source_set(Evas_Object *obj, const char *part, const char *state, double value, const char *source)
 {
-   Edje_Part_Description_Common *spd;
    Edje_Part_Description_Text *txt;
    int id_source;
 
@@ -8469,8 +8462,6 @@ edje_edit_state_text_source_set(Evas_Object *obj, const char *part, const char *
 
    if (source)
      {
-        spd = _edje_part_description_find_byname(eed, source, state, value);
-        if (!spd) return EINA_FALSE;
         txt = (Edje_Part_Description_Text *)pd;
 
         id_source = _edje_part_id_find(ed, source);
-- 
2.7.4

